src/Controller/RequestController.php

```diff
@@ -79,12 +90,18 @@ public function handleRequest($schema, $operations) {
    * @return \Drupal\Core\Cache\CacheableJsonResponse
    */
   protected function handleSingle($schema, $operations, $globals) {
-    $result = $this->processor->processQuery($schema, $operations, $globals);
+    /** @var \Drupal\graphql\GraphQL\Execution\QueryResult $result */
+    $result = [];
+    $context = new RenderContext();
+    $this->renderer->executeInRenderContext($context, function() use ($schema, $operations, $globals, &$result) {
```

Comment:
This is the wrong place, it has to be done in the field resolver.

Reply:
Not here, but within the processor and add the collected cache metadata to the result?

Reply:
First of all because we want to cache each operation individually, secondly  because we want to collect cache metadata per field, not across the whole query (partial caching at some point).

Reply:
We could also add it in the processor if we can find a good place to run that code before every field. Sure.

Reply:
@fubhy what's the most appropriate method in `FieldPluginBase`? is it or `resolveDeferred`
I am new to graphql and trying to find my ways, would something like this be considered:
```
  protected function resolveDeferred(callable $callback, $value, array $args, ResolveContext $context, ResolveInfo $info) {
    $result = NULL;
    $this->getRenderer()
      ->executeInRenderContext(new RenderContext(), function () use ($callback, $value, $args, $context, $info, &$result) {
        $result = $callback($value, $args, $context, $info);
      });
...
...
```

Reply:
Yeah, but you could even do this: 

```php
protected function resolveDeferred(callable $callback, $value, array $args, ResolveContext, $context, ResolveInfo $info) {
    $result = $this->getRenderer()
      ->executeInRenderContext(new RenderContext(), function () use ($callback, $value, $args, $context, $info, &$result) {
        return $callback($value, $args, $context, $info);
      });
}
```